src/compressor/QatAccel.cc

```diff
@@ -11,67 +11,100 @@
  * Foundation.  See file COPYING.
  *
  */
-
+#include <qatzip.h>
 #include "QatAccel.h"
 
+#define SESSIONS_NUM 200
+
+void QzSessionDeleter::operator() (struct QzSession_S *session) {
+  if (NULL != session->internal) {
+    qzTeardownSession(session);
+    qzClose(session);
+  }
+  delete session;
+}
+
 /* Estimate data expansion after decompression */
 static const unsigned int expansion_ratio[] = {5, 20, 50, 100, 200};
 
-QatAccel::~QatAccel() {
-  if (NULL != session.internal) {
-    qzTeardownSession(&session);
-    qzClose(&session);
+QatAccel::QatAccel() {
+  sessions.resize(SESSIONS_NUM);
+  for (auto &session : sessions) {
+    session.reset(new struct QzSession_S);
+    memset(session.get(), 0, sizeof(struct QzSession_S));
   }
 }
 
+QatAccel::~QatAccel() {}
+
 bool QatAccel::init(const std::string &alg) {
   QzSessionParams_T params = {(QzHuffmanHdr_T)0,};
   int rc;
 
+  std::unique_lock<std::mutex> l(lock);
```

Comment:
```suggestion
  std::unique_lock lock(mutex);
```

- avoid using `l` for variable name. it's difficult to tell it from `1` without a good font.
- C++17 is able to deduce the template parameter type.

Reply:
Accept it. `l` is similar with `1`, so changed it.